velox/functions/prestosql/tests/ArithmeticTest.cpp

```diff
@@ -177,6 +177,32 @@ TEST_F(ArithmeticTest, ln) {
   EXPECT_EQ(std::nullopt, ln(std::nullopt));
 }
 
+TEST_F(ArithmeticTest, log2) {
+  const auto log2 = [&](std::optional<double> a) {
+    return evaluateOnce<double>("log2(c0)", a);
+  };
+
+  EXPECT_EQ(log2(1), 0);
+  EXPECT_TRUE(std::isnan(log2(-1).value()));
```

Comment:
Once you rebase, you should be able to use the `IsNan()` matcher for this assertion. Here and elsewhere.
```suggestion
  EXPECT_THAT(log2(-1), IsNan());
```